synapse/storage/event_push_actions.py

```diff
@@ -100,6 +100,54 @@ def _get_unread_event_push_actions_by_room(txn):
         )
         defer.returnValue(ret)
 
+    @defer.inlineCallbacks
+    def get_push_action_users_in_range(self, min_stream_ordering, max_stream_ordering):
+        def f(txn):
+            sql = (
+                "SELECT DISTINCT(user_id) FROM event_push_actions WHERE"
+                " stream_ordering >= ? AND stream_ordering >= ?"
```

Comment:
Also, `event_push_actions` doesn't currently have an index on stream ordering? Might want to add a `(stream_ordering, user_id)` index?